### PR TITLE
NoKnockback mode

### PR DESCRIPTION
**Pull request by dastgir**

### CODE REVIEW

**Comment by 4144** [comment]
I remember last time new mode was rejected if it was non official.
May be better create plugin?

**Comment by 4144** [comment]
from other look, if it removing some hardcoded value this is good

File: src/map/skill.c [hunk 1]
```diff
@@ -2089,7 +2089,7 @@ int skill_blown(struct block_list* src, struct block_list* target, int count, in
 		case BL_MOB:
 		{
 			const struct mob_data *md = BL_UCCAST(BL_MOB, target);
-			if (md->class_ == MOBID_EMPELIUM)
+			if (md->status.mode&MD_NOKNOCKBACK)
```
Comment (MishimaHaruna) [inline src/map/skill.c]: +1 for allowing removal of hardcoded IDs

**Comment by AnnieRuru** [comment]
>I remember last time new mode was rejected if it was non official.

there are 2 monster mode is definitely official #998 
add `MD_NORANDOM_WALK` and `MD_NOCAST_SKILL` monster flag
I remember I saw a video about kro instance
that monster walk through a bridge and players have to defeat them before they pass over
these 2 monster flag is definitely official, its already in rathena

btw, can change that unsigned short mode into int ? for future expansion ...
its already in my todo list ...
or wait until my pull request turn then only change into int ?

**Comment by MishimaHaruna** [comment]
Merged via 3305409b7208befa9f5d26620d30b3490365fc21.

I also changed the mob mode field to uint32, to allow third party mods and future expandability.

**Pull request closed without merging**